### PR TITLE
Chore: Enable eslint-plugin-react partial rules

### PR DESCRIPTION
**Pull request by chihsuan**
**What this PR does / why we need it**:

This PR enables the following eslint rules in Grafana core (.eslintrc) and fixed the corresponding lint issues:

- react/display-name
- react/no-deprecated
- react/no-find-dom-node

**Which issue(s) this PR fixes**:

Partially for https://github.com/grafana/grafana/issues/29201

**Special notes for your reviewer**:

- I still use inline `// eslint-disable-next-line react/display-name` for some arrow functions since I think rewriting them by function declaration might not be a good idea for coding convention. Correct me, if I'm wrong.

### CODE REVIEW

File: packages/grafana-ui/src/components/ColorPicker/SeriesColorPickerPopover.tsx [hunk 1]
```diff
@@ -19,20 +19,22 @@ export const SeriesColorPickerPopover: FunctionComponent<SeriesColorPickerPopove
...
-              labelClass="ColorPicker__axisSwitchLabel"
-              checked={yaxis === 2}
-              onChange={() => {
-                if (onToggleAxis) {
-                  onToggleAxis();
-                }
-              }}
-            />
-          ),
+          tabComponent() {
```
Comment (dprokop) [inline packages/grafana-ui/src/components/ColorPicker/SeriesColorPickerPopover.tsx:22]: Interesting, is this related to the rules enabled in this PR?

File: packages/grafana-ui/src/components/OptionsUI/fieldColor.tsx [hunk 2]
```diff
@@ -36,7 +36,9 @@ export const FieldColorEditor: React.FC<FieldConfigEditorProps<FieldColor | unde
       description: mode.description,
       isContinuous: mode.isContinuous,
       isByValue: mode.isByValue,
-      component: () => <FieldColorModeViz mode={mode} theme={theme} />,
+      component: function render() {
+        return <FieldColorModeViz mode={mode} theme={theme} />;
+      },
```
Comment (dprokop) [inline packages/grafana-ui/src/components/OptionsUI/fieldColor.tsx:41]: Similar comment to the previous one- how does this approach help the fixes?

**Review by dprokop — APPROVED** [review]
@chihsuan thank you so much for this contribution. I went through the code and it seems to be OK! Left two questions I'm interested in getting answer for. Other than that it's awesome!

File: packages/grafana-ui/src/components/ColorPicker/SeriesColorPickerPopover.tsx (hunk 1 above)
Comment (chihsuan) [reply packages/grafana-ui/src/components/ColorPicker/SeriesColorPickerPopover.tsx:22]: Yes, this is related to the `react/display-name` rule.

Since arrow function is  anonymous, the declared component will not have the displayName. So I changed this to using shorthand method function definition to creates a named function.

File: packages/grafana-ui/src/components/OptionsUI/fieldColor.tsx (hunk 2 above)
Comment (chihsuan) [reply packages/grafana-ui/src/components/OptionsUI/fieldColor.tsx:41]: This is same like the previous one. I used regular function declaration to have the displayName.
But I just found this could be also written in the shorthand method style. I've changed this to make it consistent.  Thanks!

File: packages/grafana-ui/src/components/ColorPicker/SeriesColorPickerPopover.tsx (hunk 1 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/ColorPicker/SeriesColorPickerPopover.tsx:22]: Perfect, thanks for the explanation:)

**Comment by dprokop** [comment]
@chihsuan seems like the PR now have a conflict with the master. Can you please sync this with master and I think I can merge it if the build gets green :) Thank you!

**Comment by chihsuan** [comment]
@dprokop done. please review, thanks!
